lib/accesslists/validate.go

```diff
@@ -27,16 +27,22 @@ import (
 )
 
 // ValidateAccessListWithMembers validates a new or existing AccessList with a list of AccessListMembers.
-func ValidateAccessListWithMembers(ctx context.Context, accessList *accesslist.AccessList, members []*accesslist.AccessListMember, g AccessListAndMembersGetter) error {
-	for _, owner := range accessList.Spec.Owners {
+func ValidateAccessListWithMembers(ctx context.Context, oldAccessList, newAccessList *accesslist.AccessList, members []*accesslist.AccessListMember, g AccessListAndMembersGetter) error {
```

Comment:
I think this function's signature is quite confusing now. It takes both lists as pointers but only one of them is optional, this is easy to misuse IMO. Shall we update it to accept `newAccessList` by value to indicate that it must be non-nil?

Reply:
I'm not very keen on doing that as accesslist.AccessList is a pointer everywhere so far. Also I don't find this very confusing. Quite a typical pattern for reconciliation.